File: ACSL/Senior_2021-22/second.py


def fibCypher(option, num1, num2, key, msg):
	ans=""
	mod=26
	f=[num1%mod, num2%mod]
	for i in range(2, 202):
		f.append((f[i-1]+f[i-2])%mod)
	for i in range(1, len(f), 2):
		f[i]=-f[i]
	if(option=='E'):
		for i in range(len(msg)):
			if(i):
				ans+=" "
			k=ord(key)+f[i]
			if (k < ord('a')):
				k += 26;
			if (k > ord('z')):
				k -= 26;
			ans+=str(k * 3 + ord(msg[i]))
	else:
		nums=[int(i) for i in msg.split()]
		for i in range(len(nums)):
			k=ord(key)+f[i]
			if (k < ord('a')):
				k += 26;
			if (k > ord('z')):
				k -= 26;
			ans+=chr(nums[i]-3*k)
	return ans

print(fibCypher('D', 2, 5, 'j', "396 404 447 472 329 380 381 341 423 411 436 408 474 428 383 405 414 431 332 437 474 485 347 407 453 377 430 459 471 421 401 353 413 446 456 424 392 457 447 441 419 435 413 442 330"))
#print(fibCypher('E', 3, 7, 'h', "ACSL Sr-5 c2"))

File: ACSL/Senior_2021-22/second.cpp
#include <bits/stdc++.h>

using namespace std;


/*
 * Complete the 'fibCypher' function below.
 *
 * The function is expected to return a STRING.
 * The function accepts following parameters:
 *  1. CHARACTER option
 *  2. INTEGER num1
 *  3. INTEGER num2
 *  4. CHARACTER key
 *  5. STRING msg
 */

#define int long long
void encode(int num1, int num2, char key, string msg){
	
}

void decode(int num1, int num2, char key, string msg){

}

string fibCypher(char option, int num1, int num2, char key, string msg, const int mod=26, string ans=""){
	vector<int> f{ {num1 % mod, num2 % mod} };
	
	for (int i = 2; i < 201; i++){
		f.push_back((f[i - 1] + f[i - 2]) % mod);
	}
	for (int i = 1; i < (int)f.size(); i+=2){
		f[i] = -f[i];
	}

	if (option == 'E'){
		for (int i = 0; i < (int)msg.length(); i++){
			if (i) ans += " ";
			int ascii = (int)msg[i];
			int k = (int)key + f[i];
			if (k < 'a') k += 26;
			if (k > 'z') k -= 26;
			ans += to_string((k * 3 + ascii));
		}
	}
	else{
		
	}

	/*
	if (option == 'E'){
		encode(num1, num2, key, msg);
	}
	else decode(num1, num2, key, msg);*/
	return ans;
}

signed main(){

	cout << fibCypher('E', 3, 7, 'h', "ACSL Sr-5 c2");
	

}